### PR TITLE
Migrating IntegrationStudio to Java 11

### PR DESCRIPTION
**Pull request by Tharsanan1**
## Purpose
The IntegrationStudio 8.1.0 and older versions are not compatible with Java 11. Because of this, we cannot use eclipse 2021-06 and the latest development environment/plugins for IntegrationStudio development. There are some customer requests to use the latest features provided by eclipse 2021-12 so we need to make the IntegrationStudio source compatible with Java 11 and eclipse 2021-12.

## Goals
Make IntegrationStudio compatible with Java 11 and Eclipse 2021-12

## Approach
- Added Java 11 removed packages IntegrationStudion.kernal.library and IntegrationStudion.library.
- Changed plugin versions to Eclipse 2021-12 compatible versions. See RCP-product's feature.xml
- Added JAVA 11 JDKs to https://product-dist.wso2.com/tools/developerstudio/jdk-distributions/ and used those JDKs in the IntegrationStudio build.
- Removed "org.wso2.integrationstudio.esb.light.theme" theme from RCP plugin to make our integration studio compatible with Eclipse 2021-12. Encountered StackOverflowError when using that theme.
- removed BPS plugins from the project.

### CODE REVIEW

**Comment by rosensilva** [comment]
Fix: https://github.com/wso2/integration-studio/issues/912 with this PR

**Review by rosensilva — APPROVED** [review]

File: distribution/rcp-product/org.wso2.integrationstudio.rcp.product/integrationstudio.product [hunk 1]
```diff
@@ -52,6 +52,7 @@ Please visit : http://wso2.com/products/integration-studio/
       </win>
    </launcher>
 
+
```
Comment (ashera96) [inline distribution/rcp-product/org.wso2.integrationstudio.rcp.product/integrationstudio.product:55]: Do we need this new line?

File: components/studio-platform/features/org.wso2.integrationstudio.platform.dependencies.feature/feature.xml [hunk 2]
```diff
@@ -32,41 +32,41 @@
...
+      <plugin id="org.eclipse.ocl" download-size="0" install-size="0" version="3.17.0.v20211130-1448" unpack="false"/>
+      <plugin id="org.eclipse.ocl.common" download-size="0" install-size="0" version="1.17.0.v20211130-1448" unpack="false"/>
+      <plugin id="org.eclipse.gmf.runtime.common.ui.printing.win32" download-size="0" install-size="0" version="1.7.0.202106221344" unpack="false"/>
+      <plugin id="org.eclipse.gmf.runtime.emf.clipboard.core" download-size="0" install-size="0" version="1.7.0.202106221344" unpack="false"/>
       <plugin id="org.junit4" download-size="0" install-size="0" version="4.8.1.v20120523-1257-wso2v1" unpack="false"/>
-      <plugin id="org.eclipse.papyrus.infra.core.architecture" download-size="0" install-size="0" version="0.0.0" unpack="false"/>
+      <plugin id="org.eclipse.papyrus.infra.core.architecture" download-size="0" install-size="0" version="3.1.0.202112011019" unpack="false"/>
 
-</feature>
+</feature>
```
Comment (ashera96) [inline components/studio-platform/features/org.wso2.integrationstudio.platform.dependencies.feature/feature.xml:72]: Why have we removed the newline at the end? Was this intentional?

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/custom/deserializer/DeserializerUtils.java [hunk 3]
```diff
@@ -33,11 +33,12 @@
 import org.eclipse.gmf.runtime.notation.Node;
 import org.wso2.integrationstudio.gmf.esb.diagram.custom.AbstractConnectorEditPart;
 import org.wso2.integrationstudio.gmf.esb.diagram.custom.AbstractMediatorFlowCompartmentEditPart;
+import org.eclipse.emf.common.notify.Notifier;
```
Comment (ashera96) [inline components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/custom/deserializer/DeserializerUtils.java:36]: Shouldn't this import come before `import org.eclipse.emf.ecore.EObject;`

File: distribution/rcp-product/org.wso2.integrationstudio.rcp.product/integrationstudio.product (hunk 1 above)
Comment (Tharsanan1) [reply distribution/rcp-product/org.wso2.integrationstudio.rcp.product/integrationstudio.product:55]: Fixing : [1129](https://github.com/wso2/integration-studio/pull/1129)

File: components/studio-platform/features/org.wso2.integrationstudio.platform.dependencies.feature/feature.xml (hunk 2 above)
Comment (Tharsanan1) [reply components/studio-platform/features/org.wso2.integrationstudio.platform.dependencies.feature/feature.xml:72]: Fixing : [1129](https://github.com/wso2/integration-studio/pull/1129)

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/custom/deserializer/DeserializerUtils.java (hunk 3 above)
Comment (Tharsanan1) [reply components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/custom/deserializer/DeserializerUtils.java:36]: Fixing : [1129](https://github.com/wso2/integration-studio/pull/1129)

File: components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/SampleSimpleActivity.java [hunk 4]
```diff
@@ -1,54 +0,0 @@
-/**
- * <copyright>
```
Comment (chashikajw) [inline components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/SampleSimpleActivity.java:2]: Don't we need to add WSO2 License in these files? Please check other files as well and make the required changes if needed

File: components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/util/ExtensionSampleActivityDeserializer.java [hunk 5]
```diff
@@ -1,89 +0,0 @@
-package org.eclipse.bpel.extensionsample.model.util;
```
Comment (chashikajw) [inline components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/util/ExtensionSampleActivityDeserializer.java:1]: Shall we add add License header in these files ?

File: components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/util/ExtensionSampleActivityDeserializer.java [hunk 6]
```diff
@@ -1,89 +0,0 @@
...
-				}
-			}
-
-			return sampleStructuredActivity;
-		}
-
-		System.out.println("Cannot handle this kind of element");
-		return null;
-	}
-
```
Comment (chashikajw) [inline components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/util/ExtensionSampleActivityDeserializer.java:88]: Shall we remove extra line here ?

File: components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/util/ExtensionSampleActivitySerializer.java [hunk 7]
```diff
@@ -1,67 +0,0 @@
...
-			Activity childActivity = ((SampleStructuredActivity) activity).getActivity();
-			if (childActivity != null) {
-				activityElement.appendChild(bpelWriter.activity2XML(childActivity));
-			}
-
-			// insert the DOM element into the DOM tree
-			parentNode.appendChild(activityElement);
-		}
-	}
-
```
Comment (chashikajw) [inline components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/util/ExtensionSampleActivitySerializer.java:66]: Here as well remove the extra line

File: components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/SampleSimpleActivity.java (hunk 4 above)
Comment (Tharsanan1) [reply components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/SampleSimpleActivity.java:2]: The file is deleted and no longer in the code base.

File: components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/util/ExtensionSampleActivityDeserializer.java (hunk 6 above)
Comment (Tharsanan1) [reply components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/util/ExtensionSampleActivityDeserializer.java:88]: The file is deleted and no longer in the code base.

File: components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/util/ExtensionSampleActivitySerializer.java (hunk 7 above)
Comment (Tharsanan1) [reply components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/util/ExtensionSampleActivitySerializer.java:66]: The file is deleted and no longer in the code base.

File: components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/util/ExtensionSampleActivityDeserializer.java (hunk 5 above)
Comment (Tharsanan1) [reply components/bps-tools/plugins/examples/plugins/org.eclipse.bpel.extensionsample.model/src/org/eclipse/bpel/extensionsample/model/util/ExtensionSampleActivityDeserializer.java:1]: The file is deleted and no longer in the code base.

**Pull request merged**